synapse/rest/client/room_batch.py

```diff
@@ -123,6 +123,19 @@ async def on_POST(
                     errcode=Codes.INVALID_PARAM,
```

Comment:
> more of an [msc2716](https://github.com/matrix-org/matrix-doc/pull/2716) question: What is `prev_event_id` supposed to be when a `batch_id` is provided? If it's the insertion event associated with `batch_id`, then surely that's redundant?
>
> *-- https://github.com/matrix-org/synapse/pull/12316#pullrequestreview-924611628*

It's used as the place we're inserting next to. Which means, inheriting the `depth` and state at that point in time.

Batches of history aren't connected together in the DAG by `prev_event_id`. There is only a loose connection from the `batch` event of the next batch pointing back at the `insertion` event in the previous batch. But it's also possible to derive the same information from the previous batch but it just hasn't been that way. It can change.